### PR TITLE
ICU-13701 Optimizing increment rounding: check for nickel sooner.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-13701
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by sffc** [comment]
Incomplete: first review #237, then this should get rebased and reviewed.

**Comment by sffc** [comment]
Ready for review.

File: icu4c/source/i18n/number_rounding.cpp [hunk 1]
```diff
@@ -254,14 +262,27 @@ Precision::constructFractionSignificant(const FractionPrecision &base, int32_t m
...
     IncrementSettings settings;
+    // Note: For number formatting, fIncrement is used for RND_INCREMENT but not
+    // RND_INCREMENT_ONE or RND_INCREMENT_FIVE. However, fIncrement is used in all
+    // three when constructing a skeleton.
     settings.fIncrement = increment;
     settings.fMinFrac = static_cast<digits_t>(minFrac);
     // One of the few pre-computed quantities:
     // Note: it is possible for minFrac to be more than maxFrac... (misleading)
-    settings.fMaxFrac = roundingutils::doubleFractionLength(increment);
+    int8_t singleDigit;
```
Comment (aheninger) [inline icu4c/source/i18n/number_rounding.cpp]: Double-check that this line isn't generating compiler warnings about use of an uninitialized variable. (Even though it's safe, some compilers used to complain about things like this.)

Comment (sffc) [reply icu4c/source/i18n/number_rounding.cpp]: Confirmed that clang and gcc both are clean of warnings.

**Pull request merged**